src/archetype/createProject/SpecifyTargetFolderStep.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { Uri } from "vscode";
+import { openDialogForFolder } from "../../utils/uiUtils";
+import { IProjectCreationMetadata, IProjectCreationStep, StepResult } from "./types";
+
+export class SpecifyTargetFolderStep implements IProjectCreationStep {
+
+    public async run(metadata: IProjectCreationMetadata): Promise<StepResult> {
+        // choose target folder.
+        const result: Uri | undefined = await openDialogForFolder({
+            defaultUri: metadata.targetFolder !== undefined ? Uri.file(metadata.targetFolder) : undefined,
+            openLabel: "Select Destination Folder"
+        });
+        const targetFolder: string | undefined = result?.fsPath;
+        if (targetFolder === undefined) {
+            return StepResult.STOP;
```

Comment:
@jdneo , regarding to your comments above, if no folder is selected, the series of execution simply stop.